runners/samza/src/main/java/org/apache/beam/runners/samza/metrics/SamzaTransformMetricRegistry.java

```diff
@@ -42,26 +44,33 @@ public class SamzaTransformMetricRegistry implements Serializable {
   // TransformName -> PValue for pCollection -> Map<WatermarkId, AvgArrivalTime>
   private final ConcurrentHashMap<String, ConcurrentHashMap<String, ConcurrentHashMap<Long, Long>>>
       avgArrivalTimeMap;
+  // TransformName -> Map<WindowId, AvgArrivalTime>
+  @SuppressFBWarnings("SE_BAD_FIELD")
+  private final ConcurrentHashMap<String, ConcurrentHashMap<BoundedWindow, Long>>
```

Comment:
I think we should add some metrics to measure the size of these two maps: avgArrivalTimeMap and avgArrivalTimeMapForGbk, to prevent memory problems. The metric will be the entry count of avgArrivalTimes.

Reply:
done